### DISCUSSION TITLE
Example multi query

### DISCUSSION THREAD

**Discussion by zirill** [rusqlite/rusqlite#1104] [Questions]
Hello, please tell me.
How to execute this code in one request and get two responses at once?
```
CREATE TABLE users (name TEXT, age INTEGER);
INSERT INTO users VALUES ('1: Alice', 1);
INSERT INTO users VALUES ('2: Bob', 2);
SELECT sqlite_version();
SELECT * FROM users;
```

**Comment by gwenn** [comment]
Yon cannot.
There is nothing like https://docs.oracle.com/javase/7/docs/api/java/sql/Statement.html#getMoreResults().

**Comment by gwenn** [comment]
I forgot that in trunk, there is https://github.com/rusqlite/rusqlite/blob/master/src/lib.rs#L859-L875
So you can manually iterate on your batch.
And you can use [column_count](https://docs.rs/rusqlite/0.24.2/rusqlite/struct.Statement.html#method.column_count) to detect `SELECT`.

**Comment by zirill** [comment]
Sorry I'm a beginner and do not understand much.
I got error:
```
use rusqlite::{Batch, params, Connection, NO_PARAMS};
               ^^^^^ no `Batch` in the root
```

**Comment by gwenn** [comment]
No worry!
The `Batch` feature is not released yet. So you will have to use [rusqlite = { git = "https://github.com/rusqlite/rusqlite" }](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-git-repositories) until a new version is released.

**Comment by zirill** [comment]
Thanks a lot, it works!
```
    let sql = r"
        CREATE TABLE users (name TEXT, age INTEGER);
        INSERT INTO users VALUES ('1: Alice', 1);
        INSERT INTO users VALUES ('2: Bob', 2);
        SELECT sqlite_version();
        SELECT sqlite_version(), * FROM users limit 2;
         ";

    let mut batch = Batch::new(&conn, sql);
    let mut cix = 0;
    while let Some(mut stmt) = batch.next()? {
        cix = cix + 1;
        let mut ros = stmt.query([])?;
        while let Some(row) = ros.next().unwrap() {
            let c = row.column_count();
            println!("count: {:?}  {:?} ", cix, c);

        }
    }

```

**Comment by gwenn** [comment]
>     let mut batch = Batch::new(&conn, sql);
>     while let Some(mut stmt) = batch.next()? {

you should use `column_count` to either call `execute` or `query`
```rs
if stmt.column_count() == 0 {
   stmt.execute([]);
   continue;
}
```
>         let mut ros = stmt.query([])?;
>         while let Some(row) = ros.next().unwrap() {
>             let c = row.column_count();

**Comment by zirill** [comment]
```
let mut batch = Batch::new(&conn, sql);
while let Some(mut stmt) = batch.next()? {
```

if there is a division by teams,
can I find out the name of the current command?

```
let name = stmt.get_command_name()?;
```
(select,create,insert,..)

**Comment by gwenn** [comment]
[expanded_sql](https://docs.rs/rusqlite/0.24.2/rusqlite/struct.Statement.html#method.expanded_sql), no ?
